boring/src/ssl/mod.rs

```diff
@@ -1560,6 +1588,24 @@ pub struct CipherBits {
     pub algorithm: i32,
 }
 
+#[repr(transparent)]
+pub struct ClientHello(ffi::SSL_CLIENT_HELLO);
+
+impl ClientHello {
+    pub fn get_extension(&self, type_: u16) -> Option<&[u8]> {
+        unsafe {
+            let mut ptr = ptr::null();
+            let mut len = 0;
+            let result =
+                ffi::SSL_early_callback_ctx_extension_get(&self.0, type_, &mut ptr, &mut len);
```

Comment:
Not sure that we're dealing correctly with the memory here: we let Boring init the ptr, but it later will be dropped by Rust

Reply:
I am not sure what you mean:

> Otherwise it sets out_data to point to the extension contents (not including the type and length bytes), sets out_len to the length of the extension contents and returns one.

We borrow `self` to return the slice, so it's not like we can mutate the `ClientHello` value and bork the slice contents.

Reply:
Ah, it's const uint8_t **. I thought it's uint8_t * which we need to de-alloc ourselves.